### PR TITLE
Authentication/authorization tests

### PR DESCRIPTION
**Pull request by apedroferreira**
Tests for authentication and authorization, using the new `credentials` authentication provider that takes username/password combinations (enabled just in test mode in this PR), and some mock users.

Includes some refactors too, I saw many things to improve/fix while implementing tests.

### CODE REVIEW

File: packages/toolpad-app/src/server/auth.ts [hunk 1]
```diff
@@ -98,18 +163,45 @@ export function createAuthHandler(project: ToolpadProject): Router {
     tenantId: process.env.TOOLPAD_AZURE_AD_TENANT_ID,
   });
 
+  const credentialsProvider = CredentialsProvider({
```
Comment (apedroferreira) [inline packages/toolpad-app/src/server/auth.ts:166]: I can move the mock results somewhere else maybe, if it feels weird to have them here like this.

File: packages/toolpad-app/package.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
     }
   },
   "dependencies": {
-    "@auth/core": "0.24.0",
+    "@auth/core": "0.20.0",
```
Comment (apedroferreira) [inline packages/toolpad-app/package.json:44]: There's an issue in newer versions of `@auth/core` and I don't see any good solution proposed yet in https://github.com/nextauthjs/next-auth/issues?q=cannot+parse+action, will wait until they provide a fix or solution.

Comment (Janpot) [reply packages/toolpad-app/package.json:44]: Is there an exact issue?

File: test/playwright/tmp-0xS4E0/fixture/toolpad/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+.generated
```
Comment (Janpot) [inline test/playwright/tmp-0xS4E0/fixture/toolpad/.gitignore:1]: Accidentally committed these?

File: packages/toolpad-app/package.json (hunk 2 above)
Comment (apedroferreira) [reply packages/toolpad-app/package.json]: Nevermind it was an easy fix after all, I hadn't tried this for some reason: https://github.com/mui/mui-toolpad/pull/3056/commits/4445f9d00020f36a282cb1e29ca41bd7b53620c4

**Review by Janpot — APPROVED** [review]

**Pull request merged**